packages/components/pie-switch/test/component/pie-switch.spec.ts

```diff
@@ -1,194 +1,196 @@
+                };
+
+                await switchPage.load({ ...props });
 
                 // Act
-                await page.click(switchLabelSelector());
+                await page.getByTestId(pieSwitch.selectors.label.leading.dataTestId).click();
 
-                const pieSwitchComponent = await component.locator(componentSelector).isChecked();
+                const pieSwitchComponent = await page.getByTestId(pieSwitch.selectors.container.dataTestId).isChecked();
 
                 // Assert
                 expect(pieSwitchComponent).toBe(false);
             });
 
-            test('should emit an event with the correct name', async ({ mount, page }) => {
+            test('should emit an event with the correct name', async ({ page }) => {
                 // Arrange
-                let eventHeard = false;
-                await mount(PieSwitch, {
-                    on: {
-                        [ON_SWITCH_CHANGED_EVENT]: () => {
-                            eventHeard = true;
-                        },
-                    },
+                const switchPage = new BasePage(page, 'switch');
+                await switchPage.load();
+
+                const expectedEventMessage = 'Switch clicked';
+                const consoleMessages: string[] = [];
+
+                page.on('console', (message) => {
+                    if (message.type() === 'info') {
+                        consoleMessages.push(message.text());
+                    }
                 });
 
                 // Act
-                await page.click(componentSelector);
+                await page.getByTestId(pieSwitch.selectors.label.leading.dataTestId).click();
 
                 // Assert
-                await expect(eventHeard).toBe(true);
+                expect(consoleMessages).toEqual([expectedEventMessage]);
             });
         });
     });
 
     test.describe('Props: `aria`', () => {
         test.describe('when label exist', () => {
-            test('should render the component with the correct aria-labels', async ({ mount }) => {
+            test('should render the component with the correct aria-labels', async ({ page }) => {
                 // Arrange
                 const ariaLabelText = 'Aria label';
 
-                const component = await mount(PieSwitch, {
-                    props: {
-                        aria: {
-                            label: ariaLabelText,
-                        },
+                const switchPage = new BasePage(page, 'switch');
+                const props: Partial<SwitchProps> = {
+                    aria: {
+                        label: ariaLabelText,
                     },
-                });
+                };
+
+                await switchPage.load({ ...props });
 
                 // Act
-                const switchInput = component.locator(inputSelector);
+                const switchInput = page.getByTestId(pieSwitch.selectors.input.dataTestId);
```

Comment:
Could you modify this assertion to:


`await expect(switchInput).toHaveAttribute('aria-label', 'Aria label')`, I think it reads a bit nicer, and the added `await` with the playwright assertion will provide retry capability